### PR TITLE
Gaborlayer as nn.module

### LINKED ISSUE #6: Add poetry as dependency manager

### LINKED ISSUE #8: Add simple test case to test perfomance
When refactor, the performance (in this case, ability to learn proper image representation) should be tested. So it's necessary to find [simple dataset](https://www.kaggle.com/c/dogs-vs-cats/data) and add test case to train simple CNN on it and monitor accuracy metric.

### LINKED ISSUE #9: Remove private class inheritance
This should be removed from implementation:
```python3
from torch.nn.modules.conv import _ConvNd
from torch.nn.modules.utils import _pair

class GaborConv2d(_ConvNd):
...
```

### LINKED ISSUE #10: Remove numbers
Should have proper names:
- 3.14 / 2
- 1.41
- 8
- 0.5
- 1e-3
- 3.14

### PR DESCRIPTION
**Pull request by iKintosh**
#6 - Done
#10 - use math.pi and math.sqrt(2)
#9 - now inherit from pytorch.nn.Module
#8 - added sanity check on Dogs vs Cats dataset

### CODE REVIEW

**Comment by iKintosh** [comment]
resolves #9

**Comment by iKintosh** [comment]
resolves #10

**Comment by iKintosh** [comment]
resolves #6

**Comment by iKintosh** [comment]
resolves #8
